src/main/java/com/google/cloud/tools/appengine/cloudsdk/internal/process/DefaultProcessRunner.java

```diff
@@ -87,8 +93,14 @@ public void run(String[] command) throws ProcessRunnerException {
 
       Process process = processBuilder.start();
 
-      handleStdOut(process);
-      handleErrOut(process);
+      // Only handle stdout or stderr if there are no listeners or if output wasn't redirected,
+      // which causes process.getInputStream()/process.getErrorStream() to return NullInputStream.
+      if (!(stdOutLineListeners.isEmpty() || inheritStdOutErr)) {
```

Comment:
I think the `!stdOutLineListeners.isEmpty()` is sufficient here. There should be a check somewhere (probably the CloudSdk.Builder) to make sure that the combination of inherit=true && size(listeners) > 0 is illegal.